compiler/src/dotty/tools/dotc/core/unpickleScala2/Scala2Unpickler.scala

```diff
@@ -168,6 +168,9 @@ class Scala2Unpickler(bytes: Array[Byte], classRoot: ClassDenotation, moduleClas
   /** A mapping from method types to the parameters used in constructing them */
   private val paramsOfMethodType = new java.util.IdentityHashMap[MethodType, List[Symbol]]
 
+  /** Set of var param accessors that need to become case accessors */
+  private val paramAccessorNeedsCaseAccessor = mutable.Set.empty[Name]
```

Comment:
Hum that's not precise enough. We can read several classes at the same time, in an interleaved way, because of lazy loading. We need the map to be class-specific, i.e., be a `mutable.Map[ClassSymbol, mutable.Set[Name]]`.